src/library_webgpu.js

```diff
@@ -259,6 +266,145 @@ var LibraryWebGPU = {
+          {{{ gpu.makeGetU32('bindingPtr', C_STRUCTS.DawnBindGroupLayoutBinding.visibility) }}},
+        type: WebGPU.BindingType[
+          {{{ gpu.makeGetU32('bindingPtr', C_STRUCTS.DawnBindGroupLayoutBinding.type) }}}],
+        textureDimension: WebGPU.TextureDimension[
+          {{{ gpu.makeGetU32('bindingPtr', C_STRUCTS.DawnBindGroupLayoutBinding.textureDimension) }}}],
+        textureComponentType: WebGPU.TextureComponentType[
+          {{{ gpu.makeGetU32('bindingPtr', C_STRUCTS.DawnBindGroupLayoutBinding.textureComponentType) }}}],
+        multisampled:
+          {{{ gpu.makeGetBool('bindingPtr', C_STRUCTS.DawnBindGroupLayoutBinding.multisampled) }}},
+        // TODO this changed in upstream Dawn
+        hasDynamicOffset:
+          {{{ gpu.makeGetBool('bindingPtr', C_STRUCTS.DawnBindGroupLayoutBinding.dynamic) }}},
+      };
+    }
+
+    function makeBindings(count, bindingsPtrs) {
+      var bindings = [];
+      for (var i = 0; i < count; ++i) {
+        bindings.push(makeBinding(bindingsPtrs +
+            {{{C_STRUCTS.DawnBindGroupLayoutBinding.__size__}}} * i));
+      }
+      return bindings;
+    }
+
+    var desc = {
+      bindings: makeBindings(
+        {{{ gpu.makeGetU32('descriptor', C_STRUCTS.DawnBindGroupLayoutDescriptor.bindingCount) }}},
+        {{{ makeGetValue('descriptor', C_STRUCTS.DawnBindGroupLayoutDescriptor.bindings, '*') }}},
+      ),
+    };
+
+    var device = WebGPU.mgrDevice.get(deviceId);
+    return WebGPU.mgrBindGroupLayout.create(device.createBindGroupLayout(desc));
+  },
+
+  dawnDeviceCreateBindGroup: function(deviceId, descriptor) {
+    {{{ gpu.makeCheckDescriptor('descriptor') }}}
+
+    function makeBinding(bindingPtr) {
+      {{{ gpu.makeCheckDescriptor('bindingPtr') }}}
+
+      var bufferId = {{{ gpu.makeGetU32('bindingPtr', C_STRUCTS.DawnBindGroupBinding.buffer) }}};
+      var samplerId = {{{ gpu.makeGetU32('bindingPtr', C_STRUCTS.DawnBindGroupBinding.sampler) }}};
+      var textureViewId = {{{ gpu.makeGetU32('bindingPtr', C_STRUCTS.DawnBindGroupBinding.textureView) }}};
+#if ASSERTIONS
+      assert((buffferId != 0) + (samplerId != 0) + (textureViewId != 0) == 1);
+#endif
+
+      var binding = {{{ gpu.makeGetU32('bindingPtr', C_STRUCTS.DawnBindGroupLayoutBinding.binding) }}};
+
+      if (bufferId != 0) {
+        var size = undefined;
+
+        // Handle WGPU_WHOLE_SIZE.
+        var sizePart1 = {{{ gpu.makeGetU32('bindingPtr', C_STRUCTS.DawnBindGroupBinding.size) }}};
+        var sizePart2 = {{{ gpu.makeGetU32('bindingPtr', C_STRUCTS.DawnBindGroupBinding.size + 4) }}};
+        if (sizePart1 != 0xFFFFFFFF || sizePart2 != 0xFFFFFFFF) {
+          size = {{{ gpu.makeGetU64('bindingPtr', C_STRUCTS.DawnBindGroupBinding.size) }}};
+        }
+
+        return {
+          binding,
+          resource: {
+            buffer: WebGPU.mgrBuffer.get(bufferId),
+            offset: {{{ gpu.makeGetU64('bindingPtr', C_STRUCTS.DawnBindGroupBinding.offset) }}},
+            size,
```

Comment:
```suggestion
            size: size,
```

been trying to avoid new ES syntax because closure doesn't always like it (though this one may be ok). Also allows an emscripten build with webgpu to parse in a very old browser, which is probably desirable.